src/use-cases/queries/RecupererMesMembres.ts

```diff
@@ -0,0 +1,80 @@
+import { QueryHandler } from '../QueryHandler'
+import { UnaryOperator } from '@/shared/lang'
+
+export class RecupererMesMembres implements QueryHandler<Query, MesMembresReadModel> {
+  readonly #mesMembresLoader: MesMembresLoader
+
+  constructor(mesMembresLoader: MesMembresLoader) {
+    this.#mesMembresLoader = mesMembresLoader
+  }
+
+  async get(query: Query): Promise<MesMembresReadModel> {
+    return this.#mesMembresLoader.findMesMembres(query.codeDepartement, (mesMembres) => {
+      const membres = preFiltrageStatut(mesMembres.membres, query.statut)
+      return {
+        ...mesMembres,
+        filtres: filtresDistinct(membres),
+        membres,
+        statut: statutDistinct(mesMembres.membres),
+      }
+    })
+  }
+}
+
+export abstract class MesMembresLoader {
+  async findMesMembres(
+    codeDepartement: string,
+    filtrations: UnaryOperator<MesMembresReadModel>
+  ): Promise<MesMembresReadModel> {
+    return this.find(codeDepartement).then(filtrations)
+  }
+
+  protected abstract find(codeDepartement: string): Promise<MesMembresReadModel>
+}
+
+function preFiltrageStatut(mesMembres: MesMembresReadModel['membres'], statut: Query['statut']): MesMembresReadModel['membres'] {
```

Comment:
```suggestion
function statutsDisponibles(mesMembres: MesMembresReadModel['membres'], statut: Query['statut']): MesMembresReadModel['membres'] {
```
https://github.com/anct-cnum/suite-gestionnaire-numerique/pull/319/files#r1933475675